### ISSUE TITLE
win64 adoptOpenjdk run dacapo--tradesoap complaint java.lang.NoClassDefFoundError

### ISSUE THREAD

**Issue by yansendao** [AdoptOpenJDK/openjdk-build#1060]
**Platform:**
windows x64

**Architecture:**
x86-64

Please describe the issue that you are having with our binaries and if possible how to recreate it.

binary:https://netcologne.dl.sourceforge.net/project/dacapobench/9.12-bach-MR1/dacapo-9.12-MR1-bach.jar

run command:

D:\software\jdk\adoptOpenjdk\jdk-8.0.212.03-hotspot\bin\java.
exe -jar dacapo-9.12-bach.jar tradesoap

error log:
```
Using scaled threading model. 48 processors detected, 48 threads used to drive the workload, in a po
ssible range of [1,128]
Exception during initialization: java.lang.reflect.InvocationTargetException
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.dacapo.daytrader.Launcher.initialize(Launcher.java:70)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.dacapo.harness.Tradesoap.prepare(Tradesoap.java:45)
        at org.dacapo.harness.Benchmark.run(Benchmark.java:144)
        at org.dacapo.harness.TestHarness.runBenchmark(TestHarness.java:218)
        at org.dacapo.harness.TestHarness.main(TestHarness.java:171)
        at Harness.main(Harness.java:17)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.apache.geronimo.system.pro
perties.JvmVendor
        at org.apache.geronimo.system.logging.log4j.Log4jService.logEnvInfo(Log4jService.java:641)
        at org.apache.geronimo.system.logging.log4j.Log4jService.doStart(Log4jService.java:599)
        at org.apache.geronimo.gbean.runtime.GBeanInstance.createInstance(GBeanInstance.java:998)
        at org.apache.geronimo.gbean.runtime.GBeanInstanceState.attemptFullStart(GBeanInstanceState.
java:268)
        at org.apache.geronimo.gbean.runtime.GBeanInstanceState.start(GBeanInstanceState.java:102)
        at org.apache.geronimo.gbean.runtime.GBeanInstanceState.startRecursive(GBeanInstanceState.ja
va:124)
        at org.apache.geronimo.gbean.runtime.GBeanInstance.startRecursive(GBeanInstance.java:555)
        at org.apache.geronimo.kernel.basic.BasicKernel.startRecursiveGBean(BasicKernel.java:379)
        at org.apache.geronimo.kernel.config.ConfigurationUtil.startConfigurationGBeans(Configuratio
nUtil.java:456)
        at org.apache.geronimo.kernel.config.ConfigurationUtil.loadBootstrapConfiguration(Configurat
ionUtil.java:208)
        at org.apache.geronimo.kernel.config.ConfigurationUtil.loadBootstrapConfiguration(Configurat
ionUtil.java:167)
        at org.dacapo.daytrader.DaCapoServerRunner.initialize(DaCapoServerRunner.java:38)
        ... 14 more
```

**Comment by karianna** [comment]
Hi @yansendao Can you check with the Azul or Amazon OpenJDK distribution?  I suspect this might be a bug in Apache Geronimo as it may only accept Oracle as the vendor...

**Comment by karianna** [comment]
Actually, chances are this might be because of #1061

**Comment by smlambert** [comment]
From a quick glance, it looks like org.apache.geronimo.system.properties.JvmVendor does check System.getProperty("java.vm.vendor"), and if it does not find expected values...

else {
            // should never happen
            return new String("Unknown JVM detected - " + getFullName());
        }

So the code does seem to only look for a restricted set of vendors (sun, apache, ibm, ibmHybrid), as @karianna suggests, likely that the JvmVendor code could use an update.

**Comment by yansendao** [comment]
> Hi @yansendao Can you check with the Azul or Amazon OpenJDK distribution? I suspect this might be a bug in Apache Geronimo as it may only accept Oracle as the vendor...

Amazon seems to be OK.

![image](https://user-images.githubusercontent.com/36258307/56901262-5eec6700-6aca-11e9-872e-5b851ef3c8b6.png)

**Comment by M-Davies** [comment]
Hi @yansendao . A PR which might have resolved this (#1020) was merged so could you give this another go with one of our latest JDK's https://adoptopenjdk.net/releases.html?variant=openjdk8&jvmVariant=hotspot

**Comment by aahlenst** [comment]
Closing due to lack of requested feedback. We're happy to reopen the issue if feedback is provided by the OP (or anybody else) at a later date.
